File: app/src/main/java/nejati/me/service/listener/OnServiceStatus.java
package nejati.me.service.listener;

public interface OnServiceStatus<T> {

    /**
     * @param t the object when service have a response
     */
    void onReady(T t,int statusCode);

    /**
     * @param message  the message when service have an error
     */
    void onError(String message);
}

File: app/src/main/java/nejati/me/service/listener/OnServiceStatus.kt
package nejati.me.service.listener

interface OnServiceStatus<T> {

    /**
     * @param t the object when service have a response
     */
    fun onReady(t: T, statusCode: Int)

    /**
     * @param message  the message when service have an error
     */
    fun onError(message: String)
}
